sopel/plugins/rules.py

```diff
@@ -619,12 +619,12 @@ def match_event(self, event) -> bool:
         """
 
     @abc.abstractmethod
-    def match_intent(self, intent) -> bool:
-        """Tell if the rule matches this ``intent``.
+    def match_ctcp(self, command: Optional[str]) -> bool:
```

Comment:
Reading this before having looked at the changes to triggers, I was wondering why this would be optional. I see now that you're using `Optional` to allow the default value (`None`) of `PreTrigger.ctcp` if no CTCP command is present in the line.

One day I will get used to type annotations… I hope! 😅